### DISCUSSION TITLE
how to show file encoding in statusbar

### DISCUSSION THREAD

**Discussion by aiyolo** [LazyVim/LazyVim#672] [General]
--

**Comment by dpetka2001** [comment]
```lua
return {
  {
    "nvim-lualine/lualine.nvim",
    opts = function(_, opts)
      table.insert(opts.sections.lualine_x, "encoding")
    end,
  },
}
```
I think this should work. There's also an example in the [docs](https://www.lazyvim.org/configuration/examples) which you should read if you want to make further modifications.

**Reply by aiyolo** [reply]
I have set the following code in `example.lua`, but it still can't show the encoding of my current opend file.
```
  -- custom defined
  {
    "nvim-lualine/lualine.nvim",
    event = "VeryLazy",
    opts = function(_, opts)
      table.insert(opts.sections.lualine_x, "encoding")
    end,
  },

```

**Reply by dpetka2001** [reply]
Don't put the code in `example.lua`. The default file is set up, so it returns an empty spec.It's just there as an example for you to read. Create a file in `~/.config/nvim/lua/plugins` directory and put the code there.

**Reply by aiyolo** [reply]
thanks, it works.
